go panic in retrieval unit test

happened in ci in test with race detector
```
=== RUN   TestDelivery
--- PASS: TestDelivery (0.01s)
=== RUN   TestRetrieveChunk
=== RUN   TestRetrieveChunk/downstream
=== RUN   TestRetrieveChunk/upstream
--- PASS: TestRetrieveChunk (0.00s)
    --- PASS: TestRetrieveChunk/downstream (0.00s)
    --- PASS: TestRetrieveChunk/upstream (0.00s)
PASS
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x4afbd41]

goroutine 38 [running]:
github.com/ethersphere/bee/pkg/retrieval.(*Service).handler(0xc0000cc000, 0x4e4a880, 0xc0004ba150, 0xc000094100, 0x1, 0x8, 0x4e4d940, 0xc0000a82a0, 0x0, 0x0)
	/Users/runner/work/bee/bee/pkg/retrieval/retrieval.go:261 +0x681
github.com/ethersphere/bee/pkg/p2p/streamtest.(*Recorder).NewStream.func1(0xc0000c0240, 0xc0000be0b0, 0x4e4a840, 0xc00008e780, 0xc000094100, 0x1, 0x8, 0xc0000a82a0)
	/Users/runner/work/bee/bee/pkg/p2p/streamtest/streamtest.go:99 +0x102
created by github.com/ethersphere/bee/pkg/p2p/streamtest.(*Recorder).NewStream
	/Users/runner/work/bee/bee/pkg/p2p/streamtest/streamtest.go:96 +0x875
FAIL	github.com/ethersphere/bee/pkg/retrieval	0.569s
```

line 261 is
```go
chunk, err := s.storer.Get(ctx, storage.ModeGetRequest, swarm.NewAddress(req.Addr))
```
~so maybe `s.storer` was nil~

--- a/pkg/retrieval/retrieval_test.go
+++ b/pkg/retrieval/retrieval_test.go
@@ -172,29 +172,6 @@ func TestRetrieveChunk(t *testing.T) {
 			t.Fatalf("got data %x, want %x", got.Data(), chunk.Data())
 		}
 	})
-
-	// requesting a chunk from the upstream peer should not be possible to avoid request forwarding loops
-	t.Run("upstream", func(t *testing.T) {
-		serverAddress := swarm.MustParseHexAddress("01")
-		chunkAddress := swarm.MustParseHexAddress("02")
-		clientAddress := swarm.MustParseHexAddress("03")
-
-		server := retrieval.New(serverAddress, nil, nil, logger, accountingmock.NewAccounting(), pricer, mockValidator, nil)
-
-		recorder := streamtest.New(streamtest.WithProtocols(server.Protocol()))
-
-		clientSuggester := mockPeerSuggester{eachPeerRevFunc: func(f topology.EachPeerFunc) error {
-			_, _, _ = f(serverAddress, 0)
-			return nil
-		}}
-		client := retrieval.New(clientAddress, recorder, clientSuggester, logger, accountingmock.NewAccounting(), pricer, mockValidator, nil)
-
-		// do not request from the upstream peer
-		_, err := client.RetrieveChunk(context.Background(), chunkAddress)
-		if !errors.Is(err, topology.ErrNotFound) {
-			t.Fatalf("got error %v, want %v", err, topology.ErrNotFound)
-		}
-	})
 }
 
 type mockPeerSuggester struct {

